ethminer/MinerAux.h

```diff
@@ -309,6 +309,22 @@ class MinerCLI
 				}
 			}
 		}
+                else if (arg == "--cuda-parallel-hash")
+                {
+                        try {
+                                m_parallelHash = stol(argv[++i]);
+                                if(m_parallelHash == 0 || m_parallelHash>8)
+                                {
+                                        cerr << "Bad " << arg << " option: " << argv[i] << endl;
```

Comment:
Here you can actually throw any exception to be cached in `catch(...)`. To avoid code duplication.